plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt

```diff
@@ -173,6 +170,9 @@ internal class OrtModelBuilder : ToolingModelBuilder {
         }.filterIsInstance<ResolvedArtifactResult>()
     }
 
+    private fun AttributeContainer.getValueByName(name: String): Any? =
+        attributes.keySet().find { it.name == name }?.let { getAttribute(it) }
+
```

Comment:
Is it not possible to do `getAttribute<Any>("key")`?

Reply:
No, as the key into the container is the attribute (which consist of a name and a class type), but just the name. Which means you usually need to know the type of the attribute before you can look up its value. The whole point of the function is to be able to look up an attribute's value irrespective of the type, just by its name, as there sometimes are inconsistencies in the type an attribute uses (see the deleted comment, which is now not relevant anymore).